### PR TITLE
Add unittest for pkg/limayaml/default.yaml

### PR DESCRIPTION
**Pull request by afbjorklund**
Make sure that it is possible to parse default.yaml to the expected LimaYAML structure

Add some minor changes, to make the internal representation (before defaults) "pretty"

Left to do is to remove the default list of architectures, from the default cpuType list:

* #2333

After that, the default YAML is simply `images: []` - since that value is still mandatory.

```yaml
images: null
```

With some additional default values and "omitempty", it also matches an empty template.

```yaml
vmType: null
os: null
arch: null
images: []
cpus: null
memory: null
disk: null
mountType: null
mountInotify: null
additionalDisks: null
ssh:
  localPort: null
  loadDotSSHPubKeys: null
  forwardAgent: null
  forwardX11: null
  forwardX11Trusted: null
caCerts:
  removeDefaults: null
  files: null
  certs: null
upgradePackages: null
containerd:
  system: null
  user: null
cpuType: null
rosetta:
  enabled: null
  binfmt: null
timezone: null
firmware:
  legacyBIOS: null
audio:
  device: null
video:
  display: null
  vnc:
    display: null
networks: null
propagateProxyEnv: null
hostResolver:
  enabled: null
  ipv6: null
  hosts: null
guestInstallPrefix: null
plain: null
```

### CODE REVIEW

**Comment by afbjorklund** [comment]
The unmarshal and marshal functions should probably be refactored into separate files.

pkg/limayaml/load.go

pkg/limayaml/limayaml_test.go

**Comment by afbjorklund** [comment]
Attempt to add separate "Save" functionality:

* #2338

Also added basic test for "Validate" function:

* #2337

**Comment by AkihiroSuda** [comment]
Could you rebase and squash the commits?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: pkg/limayaml/limayaml_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"gotest.tools/v3/assert"
+)
+
+func dumpJSON(d interface{}) string {
+	b, err := json.Marshal(d)
+	if err != nil {
+		return "ERROR"
+	}
+	return string(b)
+}
```
Comment (alexandear) [inline pkg/limayaml/limayaml_test.go:17]: We can abort a test on marshal failure:
```suggestion
func dumpJSON(t *testing.T, d interface{}) string {
	b, err := json.Marshal(d)
	if err != nil {
		t.Fatal(err)
	}
	return string(b)
}
```

**Review by alexandear — APPROVED** [review]

**Review by jandubois — APPROVED** [review]

**Pull request merged**